src/verifier.rs

```diff
@@ -106,19 +143,28 @@ pub struct VerificationKey<C: Curve> {
     pub security_bits: usize,
 }
 
+/// Verifies a proof `proof` and some old proofs G points for a given verification key.
+/// If `verify_g` is `true`, the function completely verifies the proof, including the
+/// linear time check of the G point.
+/// If `verify_g` is `false`, returns an `OldProof` to be checked in a later verification.
 pub fn verify_proof_vk<C: HaloCurve, InnerC: HaloCurve<BaseField = C::ScalarField>>(
```

Comment:
By the way, would `verify_proof_vk` the preferred verification method generally since the VK is smaller? Can we get rid of `verify_proof_circuit` later?

Reply:
Yes that was the plan, or at least I've always thought that in applications we would use the verifying key. I'll keep the function for now as it can be useful for debugging, but later we can either delete it or wrap it in a `cfg(test) / cfg(debug)`.